lib/sources.js

```diff
@@ -4,60 +4,68 @@ const path = require('path');
 const fs = require('fs');
 const denodeify = require('denodeify');
 const readFile = denodeify(fs.readFile);
-const memoize = require('lodash').memoize;
 
 const polyfillsPath = path.join(__dirname, '../polyfills/__dist');
-let features = [];
+
+let features;
 let configuredAliases;
+const metadata = {};
 
 // Discover and index all the available polyfills (synchronously so that the index is available for the first request)
 try {
-	configuredAliases = require(path.join(__dirname, '../polyfills/__dist/aliases.json'));
-	features = fs
-		.readdirSync(polyfillsPath)
-		.filter(fileName => (fileName.match(/\.json$/) && fileName !== 'aliases.json'))
-		.map(fileName => fileName.replace(/\.json$/i, ''))
-	;
+	configuredAliases = require(path.join(polyfillsPath, 'aliases.json'));
+	features = fs.readdirSync(polyfillsPath).filter(f => f.indexOf('.') === -1)
+	features.forEach(featureName => {
+		metadata[featureName] = JSON.parse(fs.readFileSync(path.join(polyfillsPath, featureName, 'meta.json'), 'UTF-8'));
+	});
 } catch(e) {
+	console.log(e.stack || e);
 	throw Error("No polyfill sources found.  Run `grunt build` to build them");
 }
 
-function Collection() {
-	this.cache = {};
-}
-
-Collection.prototype.polyfillExistsSync = function(featureName) {
+exports.polyfillExistsSync = function(featureName) {
 	return (features.indexOf(featureName) !== -1);
 };
 
-Collection.prototype.listPolyfills = function() {
-	return Promise.resolve(features);
+exports.getPolyfillMetaSync = function(featureName) {
+	return metadata[featureName];
 };
 
-Collection.prototype.getPolyfill = function(featureName) {
+exports.listPolyfillsSync = function() {
+	return features;
+};
 
-	// TODO: Should this reject if the polyfill doesn't exist?  Makes the main module logic a bit more complex
-	if (features.indexOf(featureName) !== -1) {
-		if (this.cache[featureName]) {
-			return Promise.resolve(this.cache[featureName]);
-		}
-		return readFile(path.join(polyfillsPath, featureName+'.json'), 'utf-8').then(jsondata => {
-			const feature = JSON.parse(jsondata);
-			feature.name = featureName;
-			this.cache[featureName] = feature;
-			return feature;
-		});
-	} else {
-		return Promise.resolve(null);
-	}
+
+exports.listPolyfills = function() {
+	return Promise.resolve(features);
 };
 
-Collection.prototype.getConfigAliases = function(featureName) {
-	return Promise.resolve(configuredAliases[featureName]);
+exports.getConfigAliasesSync = function(featureName) {
+	return configuredAliases[featureName];
 };
 
-module.exports = {
```

Comment:
we don't offer sources as an external interface to the library... if you `require('polyfill-service')` you don't see that module, right?  Agree that we would be breaking it if you required that module individually.